terraform/production/main.tf

```diff
@@ -91,8 +75,36 @@ provider "github" {
   token = jsondecode(data.aws_secretsmanager_secret_version.flux_github_token.secret_string).flux_github_token
 }
 
-locals {
-  cluster_name = "spack-${terraform.workspace}"
-  vpc_cidr     = "10.0.0.0/16"
-  azs          = terraform.workspace == "production" ? ["us-east-1a", "us-east-1b", "us-east-1c", "us-east-1d"] : ["us-east-1a", "us-east-1b"]
+module "production_cluster" {
+  source = "../modules/spack"
+
+  deployment_name = "production"
+
+  single_nat_gateway = false
+
+  availability_zones = ["us-east-1a", "us-east-1b", "us-east-1c", "us-east-1d"]
+
+  vpc_cidr = "10.0.0.0/16"
+  public_subnets = [
+    "10.0.0.0/19",
+    "10.0.32.0/19",
+  ]
+  private_subnets = [
+    "10.0.64.0/19",
+    "10.0.96.0/19",
+    "10.0.128.0/19",
+    "10.0.160.0/19",
+  ]
+  database_subnets = [
+    "10.0.192.0/19",
+    "10.0.224.0/19",
+  ]
```

Comment:
I'm wondering if we can provide defaults for some of these variables in the module? Mainly thinking of the following

- `single_nat_gateway`
- `vpc_cidr`
- `public_subnets`
- `private_subnets`
- `database_subnets`

Don't feel too strongly about it, just a thought.

Reply:
I went back and forth on this, and ultimately decided that since we're only using this module in two places currently, the benefits of having explicit parameters all in one place outweighed the ease of reusability that having defaults would provide. I don't feel strongly either, though, especially if there's an argument for including defaults that I didn't consider.